specifyweb/frontend/js_src/lib/attachmentserverpublic.js

```diff
@@ -0,0 +1,60 @@
+"use strict";
+
+var $ = require('jquery');
+var _ = require('underscore');
+
+var schema         = require('./schema.js');
+var initialContext = require('./initialcontext.js');
+var attachmentserverbase = require('./attachmentserverbase.js');
+
+var settings;
+initialContext.load('attachment_settings.json', data => settings = data);
+
+function functionthatjustpipesthings(notused) {
+    return $.get('/attachment_gw/get_token/', { filename: 'asfdas' });
+}
+
+var attachmentserverpublic = {
+  getThumbnail: function(attachment, scale) {
+      scale || (scale = 256);
+      var style = "max-width:" + scale + "px; " + "max-height:" + scale + "px;";
+
+      var attachmentlocation = attachment.get('attachmentlocation');
+      return functionthatjustpipesthings(attachmentlocation).pipe(function(token) {
+          return $('<img>', {src: `http://localhost:5050/loris/${attachmentlocation}/full/${scale},/0/default.jpg`, style: style});
+      });
+  },
+  uploadFile: function(file, progressCB) {
+      var formData = new FormData();
+      var attachment;
+
+      formData.append('media', file);
+      var attachmentlocation = file.name;
```

Comment:
Using the user supplied filename will cause name collision problems when more than one file with same name gets uploaded. Also depending on how smart your image server is, it could open things up to directory traversal attacks and such.

You may already be planning do more here. I'm just noting it to be thorough.

Reply:
Yeah, this is indeed a problem. We are waiting to decide what the filename convention should be, so I just left it as the user uploaded filename for now.